src/dsp_tools/xmllib/models/dsp_base_resources.py

```diff
@@ -201,6 +259,25 @@ def __post_init__(self) -> None:
             case _:
                 _warn_invalid_comments(self.comments, self.res_id)
 
+    def new(
```

Comment:
I decided that the factory method only gets the mandatory fields. The optional ones they can add. I think this adds also clarity for the users with regard to mandatory fields.

Reply:
I'm not sure this is the nicest design choice, because it takes away options from the user, while I'm not sure it adds much benefit - if the optional fields have default values, then the users can simply omit them. But it's up to you.